### PR TITLE
Added Climber things to the thing (task requires climber states added to it now)

### PR DESCRIPTION
**Pull request by Frank0624789**

### CODE REVIEW

File: src/main/cpp/auton/PrimitiveParser.cpp [hunk 1]
```diff
@@ -60,6 +62,44 @@ PrimitiveParamsVector PrimitiveParser::ParseXML(string fulldirfile)
     headingOptionMap["FACE_RIGHT_STAGE"] = ChassisOptionEnums::HeadingOption::FACE_RIGHT_STAGE;
     headingOptionMap["FACE_CENTER_STAGE"] = ChassisOptionEnums::HeadingOption::FACE_CENTER_STAGE;
 
+    static std::map<std::string, noteManagerGen::STATE_NAMES> xmlStringToNOTESTATE_NAMESEnumMap{
+        {"STATE_OFF", noteManagerGen::STATE_NAMES::STATE_OFF},
+        {"STATE_READY", noteManagerGen::STATE_NAMES::STATE_READY},
+        {"STATE_FEEDER_INTAKE", noteManagerGen::STATE_NAMES::STATE_FEEDER_INTAKE},
```
Comment (witcpalek) [inline src/main/cpp/auton/PrimitiveParser.cpp:68]: delete and use noteManagerGen's public map

File: src/main/cpp/auton/PrimitiveParser.cpp [hunk 2]
```diff
@@ -190,6 +232,30 @@ PrimitiveParamsVector PrimitiveParser::ParseXML(string fulldirfile)
                             {
                                 pipelineMode = DragonCamera::PIPELINE::COLOR_THRESHOLD;
                             }
+                            else if (strcmp(attr.name(), "climberOption") == 0)
+                            {
+                                auto itr = xmlStringToCLIMBERSTATE_NAMESEnumMap.find(attr.value());
```
Comment (witcpalek) [inline src/main/cpp/auton/PrimitiveParser.cpp:237]: Add check to make sure the mechanism exists before setting this

File: src/main/cpp/auton/PrimitiveParser.cpp [hunk 3]
```diff
@@ -190,6 +232,30 @@ PrimitiveParamsVector PrimitiveParser::ParseXML(string fulldirfile)
...
+                                {
+                                    climberChosenOption = itr->second;
+                                }
+                                else
+                                {
+                                    hasError = true;
+                                }
+                            }
+                            else if (strcmp(attr.name(), "noteOption") == 0)
+                            {
```
Comment (witcpalek) [inline src/main/cpp/auton/PrimitiveParser.cpp:248]: Add check to make sure the mech already exists before setting

File: src/main/cpp/auton/ZoneParams.cpp [hunk 4]
```diff
@@ -22,13 +22,16 @@
...
-					   ChassisOptionEnums::AutonAvoidOptions)
+ZoneParams::ZoneParams(
+	int xgrid1,
+	int ygrid1,
+	int xgrid2,
+	int ygrid2,
+	ChassisOptionEnums::AutonChassisOptions,
+	ChassisOptionEnums::AutonAvoidOptions) : m_xgrid1(xgrid1),
+											 m_ygrid1(ygrid1),
+											 m_xgrid2(xgrid2),
```
Comment (witcpalek) [inline src/main/cpp/auton/ZoneParams.cpp:33]: Also set the AutonChassisOption attribute and AutonAvoidOption attribute

File: src/main/cpp/auton/ZoneParams.h [hunk 5]
```diff
@@ -35,7 +35,6 @@ class ZoneParams
         int ygrid1,
         int xgrid2,
         int ygrid2,
-        noteManagerGen::STATE_NAMES,
```
Comment (witcpalek) [inline src/main/cpp/auton/ZoneParams.h:38]: Why are we deleting the notemanager states?

File: src/main/cpp/auton/ZoneParser.cpp [hunk 6]
```diff
@@ -162,7 +131,6 @@ ZoneParams *ZoneParser::ParseXML(xml_node zonenode)
     AutonGrid::YGRID ygrid1 = AutonGrid::YGRID::NONE;
     AutonGrid::XGRID xgrid2 = AutonGrid::XGRID::NO_VALUE;
     AutonGrid::YGRID ygrid2 = AutonGrid::YGRID::NONE;
-    noteManagerGen::STATE_NAMES noteChosenOption = noteManagerGen::STATE_NAMES::STATE_OFF;
```
Comment (witcpalek) [inline src/main/cpp/auton/ZoneParser.cpp:165]: Why is this getting deleted?

File: src/main/cpp/auton/ZoneParser.cpp [hunk 7]
```diff
@@ -231,18 +199,6 @@ ZoneParams *ZoneParser::ParseXML(xml_node zonenode)
                 hasError = true;
             }
         }
-        else if (strcmp(attr.name(), "noteOption") == 0)
-        {
```
Comment (witcpalek) [inline src/main/cpp/auton/ZoneParser.cpp:235]: Why is this getting deleted?

File: src/main/cpp/auton/ZoneParser.cpp [hunk 8]
```diff
@@ -263,7 +219,6 @@ ZoneParams *ZoneParser::ParseXML(xml_node zonenode)
                                ygrid1,
                                xgrid2,
                                ygrid2,
-                               noteChosenOption,
```
Comment (witcpalek) [inline src/main/cpp/auton/ZoneParser.cpp:266]: Why is the noteManagerOption being deleted?

File: src/main/deploy/auton/auton.dtd [hunk 9]
```diff
@@ -40,16 +50,6 @@
                               		  X_U | X_V | X_W | X_X | X_Y | X_Z | X_AA ) "0"
     
     chassisOption           ( VISION_DRIVE_NOTE | VISION_DRIVE_SPEAKER | NO_VISION )
-
-    noteOption              ( STATE_OFF | STATE_READY | STATE_FEEDER_INTAKE | 
-		                      STATE_EXPEL | STATE_PLACER_INTAKE | STATE_HOLD_FEEDER_FRONT |
```
Comment (witcpalek) [inline src/main/deploy/auton/auton.dtd:45]: Why is noteOption being deleted?  It should be here as well as in Primitive

**Review by witcpalek — CHANGES_REQUESTED** [review]

File: src/main/cpp/auton/PrimitiveParser.cpp (hunk 2 above)
Comment (Frank0624789) [reply src/main/cpp/auton/PrimitiveParser.cpp:237]: Mr. Wits could you please help me with this part? I tried to create a bool variable in ClimberManagerGen/noteManagerGen and set it to true on their respective create methods, then seeing if that was true in PrimitiveParser but it popped up an error saying the variable was not a member of the class
Please helpppppp

Comment (witcpalek) [reply src/main/cpp/auton/PrimitiveParser.cpp]: Yes; I think we may have talked about this, but if not, ask me when you are working on it.

File: src/main/cpp/auton/ZoneParser.cpp [hunk 10]
```diff
@@ -114,7 +114,12 @@ ZoneParams *ZoneParser::ParseXML(xml_node zonenode)
         {"Y_53", AutonGrid::YGRID::Y_53},
         {"Y_54", AutonGrid::YGRID::Y_54}};
 
-    static std::map<std::string, noteManagerGen::STATE_NAMES> xmlStringToSTATE_NAMESEnumMap{
+    static std::map<std::string, ChassisOptionEnums::AutonChassisOptions> xmlStringToChassisOptionEnumMap{
+        {"VISION_DRIVE_NOTE", ChassisOptionEnums::AutonChassisOptions::VISION_DRIVE_NOTE},
+        {"VISION_DRIVE_SPEAKER", ChassisOptionEnums::AutonChassisOptions::VISION_DRIVE_SPEAKER},
+        {"NO_VISION", ChassisOptionEnums::AutonChassisOptions::NO_VISION},
+    };
+    static std::map<std::string, noteManagerGen::STATE_NAMES> xmlStringToNOTESTATE_NAMESEnumMap{
```
Comment (witcpalek) [inline src/main/cpp/auton/ZoneParser.cpp:122]: Use map from noteManagerGen

File: src/main/cpp/mechanisms/ClimberManager/generated/ClimberManagerGen.cpp [hunk 11]
```diff
@@ -32,6 +32,7 @@ ClimberManagerGen::ClimberManagerGen() : BaseMech ( MechanismTypes::MECHANISM_TY
 	m_stateMap()
 {
 }
```
Comment (witcpalek) [inline src/main/cpp/mechanisms/ClimberManager/generated/ClimberManagerGen.cpp:34]: Shouldn't be changing ClimberManagerGen

File: src/main/cpp/mechanisms/ClimberManager/generated/ClimberManagerGen.h [hunk 12]
```diff
@@ -49,6 +49,8 @@ class ClimberManagerGen : public BaseMech, public StateMgr
 
 	ClimberManagerGen();
```
Comment (witcpalek) [inline src/main/cpp/mechanisms/ClimberManager/generated/ClimberManagerGen.h:50]: Shouldn't be changing this file

File: src/main/cpp/mechanisms/noteManager/generated/noteManagerGen.cpp [hunk 13]
```diff
@@ -41,6 +41,8 @@ noteManagerGen::noteManagerGen() : BaseMech ( MechanismTypes::MECHANISM_TYPE::NO
 {
 }
```
Comment (witcpalek) [inline src/main/cpp/mechanisms/noteManager/generated/noteManagerGen.cpp:43]: Shouldn't be changing this file

File: src/main/cpp/mechanisms/noteManager/generated/noteManagerGen.h [hunk 14]
```diff
@@ -69,6 +69,8 @@ class noteManagerGen : public BaseMech, public StateMgr
 
 	noteManagerGen();
```
Comment (witcpalek) [inline src/main/cpp/mechanisms/noteManager/generated/noteManagerGen.h:70]: shouldn't be changing this file

**Review by witcpalek — CHANGES_REQUESTED** [review]

**Review by witcpalek — APPROVED** [review]

**Review by witcpalek — APPROVED** [review]

**Pull request merged**